modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java

```diff
@@ -149,22 +126,28 @@ public Object value() {
     @Override
     public String stringValue() {
         StringJoiner sj = new StringJoiner(", ", "{", "}");
-        String key;
-        BValue value;
-        String stringValue;
-        for (int i = 0; i < size; i++) {
-            key = "\"" + (String) values[i].getKey() + "\"";
-            value = values[i].getValue();
+
+        for (Iterator<Map.Entry<K, V>> i = map.entrySet().iterator(); i.hasNext();) {
```

Comment:
This can be done with the for-each loop, i.e. for (Map.Entry<K,V> entry : map.entrySet()) { ..